docs/guides/asterisk_installation.md

```diff
@@ -0,0 +1,182 @@
+```
+configure: Package configured for:
+configure: OS type  : linux-gnu
+configure: Host CPU : x86_64
+configure: build-cpu:vendor:os: x86_64 : pc : linux-gnu :
+configure: host-cpu:vendor:os: x86_64 : pc : linux-gnu :
+```
+
+## Set Asterisk menu options [For more options]
+
+This is one of the steps where the administrator is going to need to do his homework. There are a lot of menu options that you may or may not need. Running the following command:
+
+`make menuselect`
+
+will bring you to a [menuselect screen](images/asterisk_menuselect.png)
+
+Look through these options carefully and make selections based on your requirements. As stated earlier, this may take some additional homework.
+
+## Build and Install Asterisk
+
+To build, we want to execute the following commands in succession:
+```
+make
+make install
+```
+Installing the documentation isn't required, but unless you are a communications server expert, you'll want them installed:
+
+`make progdocs`
+
+Next install the basic PBX and make the config. The basic PBX is just that! You will probably need to make changes going forward to get your PBX to function as you want it to.
+
+```
+make basic-pbx
+make config
+```    
+
+## Asterisk Configuration
+
+### Create User & Group
+
+
+```
+groupadd asterisk
+useradd -r -d /var/lib/asterisk -g asterisk asterisk
+chown -R asterisk.asterisk /etc/asterisk /var/{lib,log,spool}/asterisk /usr/lib64/asterisk
+restorecon -vr {/etc/asterisk,/var/lib/asterisk,/var/log/asterisk,/var/spool/asterisk}
+```
+Now that the bulk of our work is completed, go ahead and exit out of the `sudo -s` command. This will require that most of the remaining commands use _sudo_ again:
+
+`exit`
+
+### Set Default User & Group
+
+`sudo vi /etc/sysconfig/asterisk`
+
+ Remove the comments on the two lines below and save:
+
+```
+AST_USER="asterisk"
+AST_GROUP="asterisk"
+```
+
+`sudo vi /etc/asterisk/asterisk.conf`
+
+Remove the comments on the two lines below and save:
+
+```
+runuser = asterisk ; The user to run as.
+rungroup = asterisk ; The group to run as.
+```
+
+### Configure Asterisk Service
+
+`sudo systemctl enable asterisk`
+
+### Configure Firewall
+
+This example uses `firewalld` for the firewall, which is the default in Rocky Linux. The goal here is to open SIP ports to the world and to open RTP (Realtime Transport Protocol) to the world on ports 10000-20000 as recommended by the Asterisk documentation. Keep in mind that you will almost certainly need other firewall rules for other forward-facing services (HTTP/HTTPS) which you will probably want to limit to your own IP addresses. That is beyond the scope of this document:
+
+```
+sudo firewall-cmd --zone=public --add-service sip --permanent
+sudo firewall-cmd --zone=public --add-port=10000-20000/udp --permanent
+```
+
+While a reboot of the server is not required to function, I've always found that this is the cleanest way to test/run after a major installation. You can choose for yourself:
```

Comment:
Based on the commands that ran above, a restart would be needed since you have created a persistent config to the firewall but have not made them effective yet or started up asterisk.

If a user did not want to restart, they could run the same firewall-cmd without `--permanent` to make the firewall changes active immediately or run `systemctl restart firewalld`.

For asterisk, if you want to enable and start asterisk all in one command you can do `systemctl enable --now asterisk`.

If you want to keep it really simple, then just have the user restart their system.